client/src/components/EditAssignmentView.js

```diff
@@ -3,41 +3,117 @@
 // SPDX-License-Identifier: MIT
 
 import React, { useState }  from 'react';
-import { useParams, useNavigate } from 'react-router-dom';
+import { useParams, useNavigate, useOutletContext } from 'react-router-dom';
 import Typography from '@mui/material/Typography';
 import Container from '@mui/material/Container';
 import Button from '@mui/material/Button';
 import Box from '@mui/material/Box';
 import Assignment from './create-assignment/Assignment';
+import ConfirmationDialog from './create-assignment/ConfirmationDialog';
 import assignmentServices from '../services/assignments';
-import mockAssignmentsClient from '../mock-data/mockAssignmentsClient';
+import mockAttainmentsClient from '../mock-data/mockAttainmentsClient';
 
 const EditAssignmentView = () => {
   const navigate = useNavigate();
-  const { assignmentId } = useParams();
-
-  // TODO: replace with a function that gets the actual data from the server
-  const [assignments, setAssignments] = useState(assignmentServices.getFinalAssignmentById(mockAssignmentsClient, Number(assignmentId)));
+  const { courseId, instanceId, sisuInstanceId, attainmentId } = useParams();
+  let addedAttainments, setAddedAttainments, attainmentIncrementId, setIncrementId;
   
+  // If this view is opened during the creation of an instance, get the necessary data from the context
+  if (sisuInstanceId) {
+    ({ addedAttainments, setAddedAttainments, attainmentIncrementId, setIncrementId } = useOutletContext());
+  }
+
+  const getAttainment = () => {
+    // If this view is opened from the course view, get attainment from DB
+    // Else the attainment is being edited during the creation of an instance so gotten from the context
+    if (instanceId) {
+      // TODO: replace with a function that gets the actual data from the server
+      return assignmentServices.getFinalAttainmentById(mockAttainmentsClient, Number(attainmentId));
+    } else if (sisuInstanceId && addedAttainments.length !== 0) {
+      let attainment = addedAttainments.find(attainment => attainment.temporaryId === Number(attainmentId));
+      if (attainment) {
+        attainment = assignmentServices.formatDates([attainment]);
+        return attainment;
+      } else {
+        console.log('Attainment could not be found');
+      }
+    } else {
+      console.log('Attainment could not be found');
+    }
+  };
+
+  const [attainments, setAttainments] = useState(getAttainment());
+
+  // Function to edit the data that is in the database
+  const editAttainment = async (attainmentObject) => {
+    try {
+      const attainment = await assignmentServices.editAttainment(courseId, instanceId, attainmentObject);
+      console.log(attainment);
+      //navigate('/' + courseId, { replace: true });
```

Comment:
This should be deleted if it's not used

Reply:
I kept this here because I will uncomment this once the get-routes for attainments work. Also, possibly add a confirmation note for the user that adding/editing worked.